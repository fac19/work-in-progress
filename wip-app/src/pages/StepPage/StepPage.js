import React from "react";
import { getFeedback } from "../../utils/get-fetch";
import FeedbackCard from "../../components/FeedbackCard/FeedbackCard";
// import StepCard from "../../components/StepCard/StepCard";

const StepPage = () => {
  const [feedback, setFeedback] = React.useState(null);
  const [isEmpty, setIsEmpty] = React.useState(false);

  const stepId = window.location.pathname.replace("/step/", "");

  React.useEffect(() => {});

  React.useEffect(() => {
    getFeedback(stepId).then((feedbackObj) => {
      setFeedback(feedbackObj);
      if (feedbackObj.length === 0) {
        setIsEmpty(true);
      }
    });
  }, [stepId]);

  const isLoading = !feedback;

  if (isLoading) {
    return <h2>Loading...</h2>;
  } else if (isEmpty) {
    return <h2>No feedback yet, add some!</h2>;
  } else {
    return (
      <section>
        <FeedbackCard feedback={feedback} />
      </section>
    );
  }
};

export default StepPage;
