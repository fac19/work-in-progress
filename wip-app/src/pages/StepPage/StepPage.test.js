import React from "react";
import TestRenderer from "react-test-renderer";
import StepPage from "./StepPage";

describe("Snapshot test", () => {
  it("renders the correct ui", () => {
    const component = TestRenderer.create(<StepPage />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
