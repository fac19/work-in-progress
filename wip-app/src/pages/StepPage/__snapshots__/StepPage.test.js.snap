// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders the correct ui 1`] = `
<h2>
  Loading...
</h2>
`;
